test/packages/parallel/awsfirehose/data_stream/log/_dev/test/pipeline/test-cloudtrail.json-expected.json

```diff
@@ -0,0 +1,28 @@
+{
```

Comment:
I guess that with the reroute processor it can happen that events are written into multiple data streams, should we take this into account and maybe have multiple expectation files, one per data stream?

Reply:
Even with the reroute processor, a single create document request will result in that document being stored in at most one data stream.

Reply:
Yes, I was wondering if we want to check that each document has been stored in the expected data stream.

Reply:
Ah, I guess I misunderstood you then. Different documents can definitely be routed to different data streams and I agree we should test that. In some sense, we're already testing it by asserting on the `data_stream` fields. Is there more you had in mind, such as checking the `_index`? That would be a bit more challenging as the `_index` field is not part of the document itself.

Reply:
Hmm even though dataset and namespace can be a list of strings, my understanding is it's either using the value of a field or the given value. For example: 
```
{
      "reroute": {
        "tag": "mydata",
        "dataset": [
          "{{container.name}}",
          "mydata"
        ]
      }
    }
```
If the `container.name` field exists in the document, it will be routed — otherwise it falls back to `mydata`.

Reply:
Yes, that's correct. For an individual document, there’s only one target data stream. The multiple values are fallbacks. But different documents can have different data streams, depending on which values are present and which if condition in the reroute processors match.

Reply:
I think it would be good to add another test document here where the condition (`if: ctx['aws.cloudwatch.log_stream'].contains('CloudTrail')`) does not match so that the document just lands in the `"awsfirehose.log"` data stream.